fe/pages/elastic_table.py

```diff
@@ -0,0 +1,475 @@
+        )
+
+    def _create_table(self, data, sort_data):
+        """Creates a Dash Bootstrap table from the provided data."""
+        if not data:
+            return html.Div("No data found.")
+
+        return dbc.Table(
+            [
+                html.Thead(
+                    html.Tr(
+                        [
+                            self._create_table_header(col, sort_data)
+                            for col in self._get_table_columns()
+                        ]
+                    )
+                ),
+                html.Tbody(
+                    [
+                        html.Tr(
+                            [
+                                html.Td(
+                                    col.display_table(row.get(col.field_name, "N/A"))
+                                )
+                                for col in self._get_table_columns()
+                            ]
+                        )
+                        for row in data
+                    ]
+                ),
+            ],
+            bordered=True,
+            hover=True,
+            responsive=True,
+        )
+
+    def table_layout(self):
+        """Returns the complete layout for the table view."""
+        default_sort_column = next(
+            (col for col in self.columns if col.default_sort), None
+        )
+
+        # Prepare filter column components
+        filter_columns = [
+            dbc.Card(
+                dbc.CardBody(
+                    [
+                        html.H5(col.display_name),
+                        dbc.Checklist(
+                            id=col.field_name,
+                            options=[],
+                            className="w-100 elastic-table-filter-checklist",
+                        ),
+                        dbc.Button(
+                            "Clear",
+                            id=f"clear-{col.field_name}",
+                            color="success",
+                            size="sm",
+                            className="mt-2 elastic-table-filter-clear",
+                        ),
+                    ]
+                )
+            )
+            for col in self._get_table_columns()
+            if col.filterable
+        ]
+
+        return html.Div(
+            [
+                # Debounce timer for input
+                dcc.Interval(
+                    id="elastic-table-timer",
+                    interval=1000,
+                    max_intervals=1,
+                    disabled=True,
+                ),
+                dbc.Row(
+                    [
+                        # Filters sidebar
+                        dbc.Col(
+                            filter_columns,
+                            width={"size": 2, "order": 1},
+                            lg=2,
+                            md=4,
+                            sm=12,
+                            xs=12,
```

Comment:
Does it make sense to have all these values? From Docs (https://dash-bootstrap-components.opensource.faculty.ai/docs/components/layout/):

Bootstrap’s grid includes six responsive tiers for building complex responsive layouts. Customize the size, order and offset of your columns on devices ranging in size from extra small to extra extra large using the xs, sm, md, lg, xl and xxl keyword arguments. Each takes the same arguments as width and specifies the column behaviour for that size of screen or larger. width secretly just sets xs. If you specify both, xs will override width.

Reply:
I simplified column width definitions, now we only have 2 parameters instead of 10: https://github.com/EMBL-EBI-ABC/PerturbationCatalogue/pull/113/commits/7e49ad270d01377c445d22c246cc1cf7f925c609